publication-commons/src/main/java/no/unit/nva/publication/model/business/PublishingRequestCase.java

```diff
@@ -200,12 +235,32 @@ public void setOwnerAffiliation(URI ownerAffiliation) {
         this.ownerAffiliation = ownerAffiliation;
     }
 
-    public Set<UUID> getApprovedFiles() {
+    public Set<File> getApprovedFiles() {
         return nonNull(approvedFiles) ? approvedFiles : Collections.emptySet();
     }
 
-    public void setApprovedFiles(Set<UUID> approvedFiles) {
-        this.approvedFiles = approvedFiles;
+    public void setApprovedFiles(Set<Object> approvedFiles) {
+        if (approvedFiles.stream().allMatch(value -> value instanceof String || value instanceof UUID)) {
+            this.approvedFiles =
+                approvedFiles.stream().map(fileIdentifier -> File.builder().withIdentifier(UUID.fromString(fileIdentifier.toString())).buildOpenFile())
+                    .collect(Collectors.toSet());
+        } else if (approvedFiles.stream().allMatch(File.class::isInstance)) {
+            this.approvedFiles =
+                approvedFiles.stream().map(map -> attempt(() -> JsonUtils.dtoObjectMapper.convertValue(map,
+                                                                                                          File.class)).toOptional())
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .collect(Collectors.toSet());
+        } else if (approvedFiles.stream().allMatch(Map.class::isInstance)) {
+            this.approvedFiles = approvedFiles.stream()
+                                     .map(map -> attempt(
+                                         () -> JsonUtils.dtoObjectMapper.convertValue(map, File.class)).toOptional())
+                                     .filter(Optional::isPresent)
+                                     .map(Optional::get)
```

Comment:
Pull this code into a deprecated utility class for ease of removal, then extract the methods that will make it possible to review the code :) I say this with kindness.